pkg/services/authn/authnimpl/service.go

```diff
@@ -286,10 +286,14 @@ func (s *Service) Login(ctx context.Context, client string, r *authn.Request) (i
 		return nil, err
 	}
 
-	namespace, id := identity.NamespacedID()
+	namespace, id := identity.GetNamespacedID()
+	intId, err := strconv.ParseInt(id, 10, 64)
+	if err != nil || intId <= 0 {
+		return nil, authn.ErrInvalidIdentityID.Errorf("expected correct identity ID but got: %s", id)
+	}
```

Comment:
We should replace these with https://github.com/grafana/grafana/blob/main/pkg/services/auth/identity/requester.go#L73

Reply:
oh nice, I totally missed this. I see there is also `UserIdentifier` which can be used in most of the places as well.

Reply:
Interesting choice by the way to put the int resolution in the Identity interface - as I was thinking to do something like this but I thought it is going to pollute the Identity interface, as parsing a possible string identity to int is not it's concern and should be done somewhere downstream. 

Curious what do you think, perhaps some helper abstraction outside of identity interface would be an alternative or something like that (not gonna change in this PR anyway)

Reply:
Yeah so this is not on the interface but just a helper function with common patterns